userland/libtock/crt1.c

```diff
@@ -15,9 +15,11 @@ extern int main(void);
 __attribute__ ((section(".start"), used))
 __attribute__ ((weak))
 __attribute__ ((noreturn))
+__attribute__ ((naked))
```

Comment:
Addressed this by just writing all of `main` in assembly. We can't allow the the compiler to touch the stack before doing the `mov r9, sp`.